src/payloads/Event.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  IsString,
+  IsOptional,
+  IsDateString,
+  IsUrl,
+  IsArray,
+  IsInt,
+  ValidateNested,
+} from 'class-validator';
+
+import { AppUser } from '@Entities';
+import { Type } from 'class-transformer';
+
+abstract class BaseEventPayload {
+  @IsString()
+  readonly name: string;
+
+  @IsString()
+  readonly description: string;
+
+  @IsString()
+  @IsOptional()
+  readonly location: string;
+
+  @IsDateString()
+  readonly startDate: string;
+
+  @IsDateString()
+  readonly endDate: string;
+
+  @IsString()
+  @IsOptional()
+  readonly type: string;
+
+  @IsUrl()
+  @IsOptional()
+  readonly rsvpURL: string;
+
+  @IsUrl()
+  @IsOptional()
+  readonly signInURL: string;
+
+  @IsUrl()
+  @IsOptional()
+  readonly fbURL: string;
+
+  @IsUrl()
+  @IsOptional()
+  readonly canvaURL: string;
+}
+
+export class EventRequest extends BaseEventPayload {
+  @IsInt({ each: true })
+  readonly hosts: number[];
```

Comment:
I thought ids for AppUsers are strings? Or are they just numbers in string form?

Reply:
postgres uses autoincrement integers instead of uuids, that's why all ids are numbers

Reply:
aaaaaaaaaaaaaand you're right I trolled - good catch @thaigillespie

Reply:
haha npnp